test/e2e/util_test.go

```diff
@@ -822,6 +817,114 @@ func verifyInternalIngressController(t *testing.T, name types.NamespacedName, ho
 	}
 }
 
+// waitForLookupIP attempts to resolve the IP address for a given hostname within a specified timeout period.
+// It handles platform-specific DNS requirements, particularly for environments that may need internal cluster
+// DNS resolution or a DNS warmup period to avoid negative caching (like IBMCloud). It returns the first IP
+// address from the resolved list of IPs.
+func waitForLookupIP(t *testing.T, host string, timeout time.Duration, forceInternal bool) (string, error) {
+	// If already an IP, return as is.
+	if net.ParseIP(host) != nil {
+		return host, nil
+	}
+
+	var ip []net.IP
+	var err error
+	if err := wait.PollUntilContextTimeout(context.Background(), 10*time.Second, timeout, true, func(ctx context.Context) (bool, error) {
+		warmup, needsInternalDNSResolution := platformsNeedInternalDNSResolutionAndWarmup[infraConfig.Status.PlatformStatus.Type]
+		if needsInternalDNSResolution || forceInternal {
+			if warmup > 0 {
+				t.Logf("this platform requires DNS warmup time to avoid negative caching...waiting for %s", warmup)
+				time.Sleep(warmup)
```

Comment:
I'm wondering how sleeping for 7 minutes live together with the polling timeout of 5 minutes. Since the ibm test passed I guess that `PollUntilContextTimeout` checks the timeout only after the given function returns (and if it doesn't return `true`).

Reply:
Good catch, you are correct, the loop still works as long as it is resolved on the first try, since our 7 min warmup period burnt up our loop timeout of 5 minutes. That's fragile.

I'll move the warmup period outside of the polling loop.